File: BB.Core/Model/UserDetails.cs
using System;

namespace BB.Core.Model
{
    public class UserDetails
    {
        public long UserID { get; set; }

        public string FirstName { get; set; }

        public string MiddleName { get; set; }

        public string LastName { get; set; }

        public Gender Gender { get; set; }

        public DateTime Birthday { get; set; }

        public DateTime RegistrationDate { get; set; }

        public string PhoneNumber { get; set; }

        public Address Address { get; set; }

        public virtual User User { get; set; }
    }
}

File: BB.Web/app/Model/Users/UserDetails.ts
import u = require("./User");
import User = u.User;
import g = require("../Enum/Gender");
import Gender = g.Gender;
import a = require("../Address/Address");
import Address = a.Address;
export class UserDetails
{
    public UserID: number;
    public FirstName: string;
    public MiddleName: string;
    public LastName: string;
    public Gender: Gender;
    public Birthday: Date;
    public PhoneNumber: string;
    public Address: Address;
    public User: User;
}